### PR TITLE
Remove empty result files by default

### PR DESCRIPTION
**Pull request by marwatk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -330,38 +330,43 @@ a step further by templating many files in a single run. The `template`
 function's `--help` provides examples:
 
 ```bash
-This will take an arbitrary number of source templates (or folders full
-of templates) and process them either in place (see --in-place) or into the
-folder specified as the last argument. It will make any folders required
-along the way. If a source is an existing file (not a folder) it will be
-treated as a template regardless of the extension (though if the extension
-matches it will still be removed).
+$ go run . template --help
```
Comment (lucastheisen) [inline README.md]: I don't think there is any value in the readme showing how to run the `--help`

File: clconf/template.go [hunk 2]
```diff
@@ -125,17 +127,27 @@ func processTemplate(paths pathWithRelative, dest string, value interface{},
...
-	err = ioutil.WriteFile(target, []byte(content), mode)
-	if err != nil {
-		return result, err
-	}
-
-	if !options.KeepExistingPerms {
-		// WriteFile only uses the mode if the file is created during write
-		err = os.Chmod(target, mode)
+	if options.KeepEmpty || content != "" {
+		// This won't change the mode if it already exists
```
Comment (lucastheisen) [inline clconf/template.go]: this comment doesn't help.  the comment where you run another chmod basically tells the story.

File: clconf/template.go [hunk 3]
```diff
@@ -125,17 +127,27 @@ func processTemplate(paths pathWithRelative, dest string, value interface{},
...
+			// and even then filters on the umask so we os.Chmod to get the
+			// requested perms.
+			err = os.Chmod(target, mode)
+			if err != nil {
+				return result, err
+			}
+		}
+	} else { //Don't keep the empty result
+		_, err = os.Stat(target)
+		if err == nil {
```
Comment (lucastheisen) [inline clconf/template.go]: Pretty sure this is dangerous... IIRC, there is more than just _file not found_ error possible here.  I am pretty sure you are supposed to do something like:
```go
if err == nil || err != os.FileNotFoundError
```
or something similar...  but i guess remove might be fine with the file not existing?

File: clconf/template.go [hunk 4]
```diff
@@ -125,17 +127,27 @@ func processTemplate(paths pathWithRelative, dest string, value interface{},
...
+			os.Remove(target)
```
Comment (lucastheisen) [inline clconf/template.go]: Dont wanna check this error?  Should clconf template processing report a failure if it leaves the file in place?

File: README.md (hunk 1 above)
Comment (marwatk) [reply README.md]: [Fixed](https://github.com/pastdev/clconf/pull/13/commits/e86bd2cdf541c247e3007ed37169507d2d56a096)

File: clconf/template.go (hunk 3 above)
Comment (marwatk) [reply clconf/template.go]: [Fixed](https://github.com/pastdev/clconf/pull/13/commits/e86bd2cdf541c247e3007ed37169507d2d56a096)

File: clconf/template.go (hunk 4 above)
Comment (marwatk) [reply clconf/template.go]: [Fixed](https://github.com/pastdev/clconf/pull/13/commits/e86bd2cdf541c247e3007ed37169507d2d56a096)

File: clconf/template.go (hunk 2 above)
Comment (marwatk) [reply clconf/template.go]: [Fixed](https://github.com/pastdev/clconf/pull/13/commits/e86bd2cdf541c247e3007ed37169507d2d56a096)

**Pull request merged**